src/SDKConnector.php

```diff
@@ -250,8 +250,16 @@ private function buildCredentials(KeyInterface $key): CredentialsInterface {
    */
   protected function userAgentPrefix(): string {
     if (NULL === self::$userAgentPrefix) {
-      $userAgent = \Drupal::moduleHandler()->invokeAll('user_agent_string_alter');
-      $userAgent = implode('; ', $userAgent);
+      // apigee_edge module info.
+      $edge_module_info = $this->infoParser->parse($this->moduleHandler->getModule('apigee_edge')->getPathname());
+      if (!isset($edge_module_info['version'])) {
+        $edge_module_info['version'] = '2.x-dev';
+      }
+      $user_agent[] = $edge_module_info['name'] . '/' . $edge_module_info['version'];
+
+      // Get info from other modules.
+      $userAgent = \Drupal::moduleHandler()->invokeAll('apigee_edge_user_agent_string', [&$user_agent]);
+      $userAgent = !empty($userAgent) ? implode('; ', $userAgent) : implode('; ', $user_agent);
 
       self::$userAgentPrefix = $userAgent . '; Drupal/' . \Drupal::VERSION;
```

Comment:
It is still looks odd that the module version can be removed from the user agent string via an alter hook (yes, that is possible) but not the Drupal version suffix.
Maybe the alter hook should allow altering the complete string (e.g: `Apigee Edge/2.x-dev;Drupal/8.9.10`) and in an alter hook implementation the string should be parsed, exploded and imploded again as needed. (We intentionally introduced a format that can be easily parsed ;) ).

Reply:
@shishir-intelli First I think we should get an agreement on this because other suggested changes could become unnecessary :)

Reply:
The other option is the alter hook still gets an array of user agent string parts (Component and version) that includes Apigee Edge version and Drupal version by default and a hook implementation can add new items to that array in any position, or even remove items. The user agent string would be still built in the SdkConnector to the expected format. (Probably that is the better option due to the previous sentence)

Reply:
Yes, By default Apigee Edge version and Drupal version in array and in hook implementation one can add new items or update the position, In our case m10 module info is placed at beginning of array. so now the format looks like 
`Apigee Monetization/8.x-1.11; Apigee Edge/8.x-1.24; Drupal/9.2.6 (Apigee Edge PHP Client 2.0.10; PHP/# )`